src/NUnitFramework/framework/Assert.cs

```diff
@@ -96,13 +102,20 @@ public static new void ReferenceEquals(object a, object b)
         /// </summary>
         /// <param name="message">The message to initialize the <see cref="AssertionException"/> with.</param>
         /// <param name="args">Arguments to be used in formatting the message</param>
-        static public void Pass(string message, params object[] args)
+        public static void Pass(string message, params object[] args)
         {
-            if (message == null) message = string.Empty;
-            else if (args != null && args.Length > 0)
-                message = string.Format(message, args);
+            Assert.Pass(BuildExceptionMessageFunc(message, args));
+        }
 
-            throw new SuccessException(message);
+        /// <summary>
+        /// Throws a <see cref="SuccessException"/> with the message and arguments 
+        /// that are passed in. This allows a test to be cut short, with a result
+        /// of success returned to NUnit.
+        /// </summary>
+        /// <param name="getExceptionMessage">A function to build the message included with the Exception</param>
+        public static void Pass(Func<string> getExceptionMessage)
```

Comment:
There's no purpose to this as far as I can see. The reason for using a Func is to avoid extra work when the message is not actually needed. With Assert.Pass, Assert.Fail, Assert.Inconclusive and Assert.Ignore it's always used.

If we leave out those changes, do we need to change this file at all? Is the BuildExceptionMessageFunc used from other Assert files?